### PR TITLE
Added bid pool and fixed getAllWinningBids function

### PR DESCRIPTION
**Pull request by jaiminpanchal27**
## Type of change
- [x] Bugfix
- [x] Feature

## Description of change
This PR adds bid pool concept to prebid.js 
In case of concurrent auctions, auctionManager can have multiple bids from same bidder for a particular adUnit. While selecting bids to do auction, prebid.js uses following criteria if such multiple bids are found.
- Do not use bid which is rendered or won earlier auction.
- Select bid whose `ttl` is not expired. Here buffer of ~~200~~ 1000 ms is included in check.
- Use first in first out. Select the bid which came in first for auction.

Also added new api function `getAllPrebidWinningBids` which will return the bids won in prebid auction but not rendered by dfp.

Also fixed the bug in `getAllWinningBids` described here #2238

### CODE REVIEW

File: src/targeting.js [hunk 1]
```diff
@@ -10,14 +10,29 @@ var CONSTANTS = require('./constants.json');
 var pbTargetingKeys = [];
 
 export const BID_TARGETING_SET = 'targetingSet';
+export const RENDERED = 'rendered';
 
 const MAX_DFP_KEYLENGTH = 20;
+const TTL_BUFFER = 200;
```
Comment (mkendall07) [inline src/targeting.js]: let's make the buffer 1000

File: src/auction.js [hunk 2]
```diff
@@ -197,8 +197,8 @@ export function newAuction({adUnits, adUnitCodes, callback, cbTimeout, labels})
     executeCallback,
     callBids,
     bidsBackAll,
-    setWinningBid: (winningBid) => { _winningBid = winningBid },
-    getWinningBid: () => _winningBid,
+    setWinningBid: (winningBid) => { _winningBids = _winningBids.concat(winningBid) },
```
Comment (snapwich) [inline src/auction.js]: Seems `addWinningBid` is a more apt name now.

File: src/auction.js [hunk 3]
```diff
@@ -197,8 +197,8 @@ export function newAuction({adUnits, adUnitCodes, callback, cbTimeout, labels})
...
+    getWinningBid: () => _winningBids,
```
Comment (snapwich) [inline src/auction.js]: and `getWinningBids` here.

**Comment by jaiminpanchal27** [comment]
Docs PR here https://github.com/prebid/prebid.github.io/pull/700

**Pull request merged**